Content.Client/Doors/DoorSystem.cs

```diff
@@ -131,6 +131,18 @@ private void OnAppearanceChange(EntityUid uid, DoorComponent comp, ref Appearanc
             default:
                 throw new ArgumentOutOfRangeException($"Invalid door visual state {state}");
         }
+
+        // if we have https://github.com/space-wizards/RobustToolbox/pull/3984
+        // if (args.Sprite.LayerExists(DoorVisualLayers.BaseUnlit))
+        if (args.Sprite.LayerMapTryGet(DoorVisualLayers.BaseUnlit, out _, false))
```

Comment:
In general just use `LayerMapTryGet()` and pass the layer index to `LayerSetWhatever()`